Implement VSCode providers

rdurfee opened the pull request:
Implement VSCode language providers for Ren'Py

LuqueDaniel commented:
Thanks for your contribution! It will take a while to review everything, but I will start this weekend.

rdurfee commented:
Let me know if you have any questions.

LuqueDaniel reviewed (changes requested):
Thanks for the work!

I have identified some points that could be improved:

- Remove commented code that will not be used
- Modularize the code for easier readability and maintainability. Files like `src/navigationdata.ts` or `src/extension.ts` are very long and it might be a good idea to split them up.
- Implement each feature in a different commit. This is not really necessary and can also cause trouble.

Feel free to discuss any of these points or the comments of the review.

rdurfee commented:
Thanks for the notes. I'll check them out and make changes as needed.

rdurfee commented on src/renpy.json:
```diff
@@ -0,0 +1 @@
+{"renpy": {"renpy.count_seen_dialogue_blocks": ["internal", "function", "()", "", "", "Returns the number of dialogue blocks the user has seen in any play-through of the current game."], "renpy.timeout": ["internal", "function", "(seconds)", "", "", "Causes an event to be generated before `seconds` seconds have elapsed. This ensures that the event method of a user-defined displayable will be call ...
```
We can reformat if necessary for easy editing. The file is actually auto-generated via a hack I made to the sphinx project that's part of the Ren'Py SDK. I hope to get Tom to build it into sphinx.

LuqueDaniel replied:
I understand since it's generated it isn't necessary to reformat it. It would be good to somehow document the process to regenerate this file from your hack.

rdurfee commented:
I submitted an updated commit. I moved most of the code out of extension.ts and created separate files for each provider. I also moved the related code from NavigationData into the provider files, so it's much more manageable now.

rdurfee commented:
@LuqueDaniel  I fixed the issue with the edits stripping the quotes, and it also matches single quotes now. (I did not allow matches for backquotes, because RenPy will not compile when using backquotes.) Thanks!

LuqueDaniel commented:
Thanks, I'll try to disable some extensions.

rdurfee commented:
@LuqueDaniel I've pushed an updated README.md file!

LuqueDaniel commented:
@rdurfee Thank you for adding those features and correcting my typos. Could you have accidentally added the `Visual Studio Code.edit.py` file?

LuqueDaniel commented:
> 
> 
> @LuqueDaniel Do you want me to remove it? Hopefully at some point, Code will get added to RenPy's Text Editor list.

Don't worry, I'll try to add it as a note in the `README.md ` 🙂
